### PR TITLE
[IMP] l10n_cl_etd_account: Set the correct SII document when refunding

### PR DESCRIPTION
**Pull request by max3903**

### CODE REVIEW

**Review by nelsonramirezs — APPROVED** [review]
LGTM

File: l10n_cl_etd_account/models/account_invoice.py [hunk 1]
```diff
@@ -6,6 +6,17 @@
...
 
 
+SII_MAPPING = {
+    # class_id.code: class_id.code of the refund
+    30: 60,  # Factura -> Nota de Crédito
+    33: 61,  # Factura Electrónica -> Nota de Crédito Electrónica
+    35: 60,  # Boleta -> Nota de Crédito
+    39: 61,  # Boleta Electrónica -> Nota de Crédito Electrónica
+    60: 55,  # Nota de Crédito -> Nota de Débito
+    61: 56,  # Nota de Crédito Electrónica -> Nota de Débito Electrónica
```
Comment (dreispt) [inline l10n_cl_etd_account/models/account_invoice.py:16]: This is good as a first approach, but the mapping could be done in the `sii.document-class`` Model.

File: l10n_cl_etd_account/models/account_invoice.py [hunk 2]
```diff
@@ -29,3 +40,26 @@ def invoice_validate(self):
...
+        return code and SII_MAPPING.get(code, False) or False
+
+    def get_reverse_sii_document(self):
+        sii_code = self.get_reverse_sii_code(code=self.class_id.code)
+        return self.env['sii.document.class'].search([
+            ('code', '=', sii_code)
+        ], limit=1)
+
+    @api.multi
+    @api.returns('self')
```
Comment (dreispt) [inline l10n_cl_etd_account/models/account_invoice.py:54]: Why the api.returns decorator? Could it be removed?

File: l10n_cl_etd_account/models/account_invoice.py [hunk 3]
```diff
@@ -29,3 +40,26 @@ def invoice_validate(self):
...
+    def refund(self, date_invoice=None, date=None, description=None,
+               journal_id=None):
+        refunds = super().refund(
+            date_invoice=date_invoice, date=date, description=description,
+            journal_id=journal_id)
+        index = 0
+        for invoice in self:
```
Comment (dreispt) [inline l10n_cl_etd_account/models/account_invoice.py]:
```suggestion
        for index, invoice in enumerate(self):
```

File: l10n_cl_etd_account/wizard/account_invoice_refund.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+class AccountInvoiceRefund(models.TransientModel):
+    _inherit = 'account.invoice.refund'
+
+    @api.multi
+    def compute_refund(self, mode='refund'):
+        res = super().compute_refund(mode)
+        if mode == 'modify':
+            inv_obj = self.env['account.invoice']
+            context = dict(self._context or {})
```
Comment (dreispt) [inline l10n_cl_etd_account/wizard/account_invoice_refund.py:14]: variable set but not used
```suggestion
```

**Review by dreispt — APPROVED** [review]

File: l10n_cl_etd_account/models/account_invoice.py (hunk 2 above)
Comment (max3903) [reply l10n_cl_etd_account/models/account_invoice.py:54]: It is there on the original method...

**Review by nelsonramirezs — APPROVED** [review]

File: l10n_cl_etd_account/models/account_invoice.py [hunk 5]
```diff
@@ -29,3 +40,24 @@ def invoice_validate(self):
...
+        ], limit=1)
+
+    @api.multi
+    @api.returns('self')
+    def refund(self, date_invoice=None, date=None, description=None,
+               journal_id=None):
+        refunds = super().refund(
+            date_invoice=date_invoice, date=date, description=description,
+            journal_id=journal_id)
+        for index, invoice in self:
```
Comment (max3903) [inline l10n_cl_etd_account/models/account_invoice.py]:
```suggestion
        for index, invoice in enumerate(self):
```

**Review by nelsonramirezs — APPROVED** [review]

**Review by nelsonramirezs — COMMENTED** [review]
When partner.invoicing_policy == 'invoice':
Could be sii_code = 33 or 34.
33 Is an Invoice with IVA tax and 34 without or exempt

**Comment by max3903** [comment]
@nelsonramirezs How do you handle it? Is it determined by the fiscal position or you check if there is a tax line?

What about 41 / Boleta Electronica Exenta?

**Comment by nelsonramirezs** [comment]
> @nelsonramirezs How do you handle it? Is it determined by the fiscal position or you check if there is a tax line?
> 
Never handled this before. But check if there is a tax line is good idea. 

> What about 41 / Boleta Electronica Exenta?
For Boleta Electronica Exenta also.

**Comment by max3903** [comment]
@nelsonramirezs Fixed.

**Review by nelsonramirezs — APPROVED** [review]
Love it

**Review by nelsonramirezs — APPROVED** [review]

**Review by dreispt — APPROVED** [review]

**Pull request merged**